docs/general/dql/selects.rst

```diff
@@ -740,6 +740,50 @@ an upper-bound::
     Both the ``from`` and ``to`` index values are inclusive.
     Using an index greater than the array size results in an empty array.
 
+For selecting records having values with empty or unassigned arrays, like::
+
+    cr> insert into locations (id, name, position, kind, landmarks) values
+    ... (99, 'Voidspace', 0, 'Milliways', []);
+    INSERT OK, 1 row affected (... sec)
+
+.. Hidden: refresh locations
+
+    cr> refresh table locations;
+    REFRESH OK, 1 row affected (... sec)
+
+a suitable expression to match for both ``NULL`` values and empty arrays, is::
```

Comment:
I think if we document this like that we'd have to explain why empty arrays match on `IS NULL`. (Is that even still the case?)

Reply:
> Is that even still the case?

From my experiments, and as implemented here as a doctest, I think it is still the case, yes. However, for empty or null arrays, it apparently only works on real tables.

```sql
cr> select array_length([], 1);
SQLParseException[The inner type of the array argument `array_length` function cannot be undefined]
cr> select array_length(null, 1);
SQLParseException[The inner type of the array argument `array_length` function cannot be undefined]
```

> If we document this like that we'd have to explain why empty arrays match on `IS NULL`.

Yes, this also was not obvious to me and I only found out by experimenting and reading the issue tracker (https://github.com/crate/crate/issues/9398#issuecomment-559750730). If you think it makes sense, I will be happy if you could drop some words about it so I can carry it over to the documentation.